src/app/annotationViewer/positionLine.js

```diff
@@ -5,121 +5,123 @@ angular
         return {
             restrict: "A",
             templateUrl: "annotationViewer/positionLine.tpl.html",
-            replace: true,
+            replace: false,
             scope: {
-                media: "<media",
-                audioData: "<audioData",
-                // TODO: remove this binding
-                imageClass: "<imageClass"
+                media: "=media",
+                audioData: "=audioData"
             },
-            link: function (scope, elements, attributes, controller) {
+            controller: ["$scope", "$element", "lodash", function (scope, element, _) {
 
-                // the dom element for the spectrogram image that is used to get
-                // the width to calculate the pixel offset for the position line
-                // TODO: remove the need to reference an image element.
-                var image;
+                var self = this;
+                self.containerWidth = 0;
+
+                self.updateContainerWidth = function () {
+                    var container = element[0].parentElement;
+                    self.containerWidth = (container.clientWidth > 0) ? container.clientWidth : 1;
 
-                /**
-                 * Searches for the closest node to the position-line element that
-                 * has the class scope.imageClass
-                 * If there are multiple spectrograms on the page, for example generated in a loop,
-                 * this ensures that each position line is using the correct spectrogram image.
-                 * Searches for the class in the parent dom element, and if not found recurses up one level of dom and repeats.
-                 * @param element; the current element in the recursion that is being searched
-                 * @param depth; int, the current level of recursion, in order to limit the search
-                 * @returns DOM element or NULL
-                 * @TODO: deprecated. We don't want to be searching for DOM elements here. Better to bind to data.
-                 *
-                 */
-                this.getImageElement = function (element, depth) {
-                    if (depth > 3 || element.tagName === "BODY") {
-                        return null;
-                    }
-                    var containingElement = element.parentElement;
-                    var images = containingElement.getElementsByClassName(scope.imageClass);
-                    if (images.length > 0) {
-                        return images[0];
-                    } else {
-                        return this.getImageElement(containingElement, depth + 1);
-                    }
                 };
 
+                self.getWidth = function () {
+                    if (self.containerWidth < 2) {
```

Comment:
It was to determine whether the containerWidth had been set based on clientWidth after the elements had finished rendering. It was overly complicated and has been simplified.